src/System.Linq/src/System/Linq/Skip.cs

```diff
@@ -121,5 +122,43 @@ private static IEnumerable<TSource> SkipWhileIterator<TSource>(IEnumerable<TSour
                 }
             }
         }
+
+#if netcoreapp11
+        public static IEnumerable<TSource> SkipLast<TSource>(this IEnumerable<TSource> source, int count)
+        {
+            if (source == null)
+            {
+                throw Error.ArgumentNull(nameof(source));
+            }
+
+            if (count <= 0)
+            {
+                return source.Skip(0);
+            }
+
+            return SkipLastIterator(source, count);
+        }
+
+        private static IEnumerable<TSource> SkipLastIterator<TSource>(IEnumerable<TSource> source, int count)
+        {
+            Debug.Assert(source != null);
+            Debug.Assert(count > 0);
+
+            var queue = new Queue<TSource>();
+
+            using (IEnumerator<TSource> e = source.GetEnumerator())
+            {
+                while (e.MoveNext())
+                {
```

Comment:
@JonHanna You are correct, that seems like it would allow for a tighter loop (and a slightly faster comparison too, because now 0 is a comparand). I'll use your implementation in the iterator.